File: sdks/java/src/main/java/com/puxbay/resources/CustomersResource.java
package com.puxbay.resources;

import com.puxbay.Puxbay;
import com.puxbay.exceptions.PuxbayException;
import com.puxbay.models.PaginatedResponse;
import com.puxbay.models.Customer;

public class CustomersResource {
    private final Puxbay client;

    public CustomersResource(Puxbay client) {
        this.client = client;
    }

    public PaginatedResponse<Customer> list(int page, int pageSize) throws PuxbayException {
        return (PaginatedResponse<Customer>) client.request("GET", "customers/?page=" + page + "&page_size=" + pageSize, null, PaginatedResponse.class);
    }

    public Customer get(String customerId) throws PuxbayException {
        return client.request("GET", "customers/" + customerId + "/", null, Customer.class);
    }

    public Customer create(Customer customer) throws PuxbayException {
        return client.request("POST", "customers/", customer, Customer.class);
    }

    public Customer update(String customerId, Customer customer) throws PuxbayException {
        return client.request("PATCH", "customers/" + customerId + "/", customer, Customer.class);
    }

    public void delete(String customerId) throws PuxbayException {
        client.request("DELETE", "customers/" + customerId + "/", null, Void.class);
    }
}

File: sdks/kotlin/src/main/kotlin/com/puxbay/resources/CustomersResource.kt
package com.puxbay.resources

import com.puxbay.Puxbay
import com.puxbay.models.Customer
import com.puxbay.models.PaginatedResponse
import io.ktor.http.HttpMethod

class CustomersResource(client: Puxbay) : BaseResource(client) {
    suspend fun list(page: Int = 1, pageSize: Int = 20): PaginatedResponse<Customer> =
        client.request(HttpMethod.Get, "customers/?page=$page&page_size=$pageSize")
    
    suspend fun get(customerId: String): Customer =
        client.request(HttpMethod.Get, "customers/$customerId/")
    
    suspend fun create(customer: Customer): Customer =
        client.request(HttpMethod.Post, "customers/", customer)
    
    suspend fun update(customerId: String, customer: Customer): Customer =
        client.request(HttpMethod.Patch, "customers/$customerId/", customer)
        
    suspend fun delete(customerId: String) {
        client.request<Unit>(HttpMethod.Delete, "customers/$customerId/")
    }
        
    suspend fun adjustLoyaltyPoints(customerId: String, points: Int, description: String): Customer =
        client.request(HttpMethod.Post, "customers/$customerId/adjust-loyalty-points/", 
            mapOf("points" to points, "description" to description))
}
